source/config/featureFlag.py

```diff
@@ -0,0 +1,226 @@
+			f"{OptionsEnumClass.__qualname__} (got {value})"
+		)
+
+	return FeatureFlag(value, behaviorOfDefault)
+
+
+def _transformSpec_AddFeatureFlagDefault(specString: str, **kwargs) -> str:
+	""" Ensure that default is specified for featureFlag used in configSpec.
+	Param examples based on the following spec string in configSpec:
+		loadChromiumVBufOnBusyState = featureFlag(behaviorOfDefault="enabled", optionsEnum="BoolFlag")
+	@param specString: EG 'featureFlag(behaviorOfDefault="enabled", optionsEnum="BoolFlag")'
+	@param kwargs: EG {'behaviorOfDefault': 'enabled', 'optionsEnum':'BoolFlag'}
+	@return 'featureFlag(behaviorOfDefault="ENABLED", optionsEnum="BoolFlag", default="DEFAULT")'
+	@remarks Manually specifying 'default' in the configSpec string (for featureFlag) will result in a
+		VdtParamError. Required params:
+		- 'behaviorOfDefault'
+		- 'optionsEnum'
+	"""
+	log.info(f"specString: {specString}, kwargs: {kwargs}")
+	usage = 'Usage: featureFlag(behaviorOfDefault="enabled"|"disabled", optionsEnum="BoolFlag")'
+	if "default=" in specString:
+		raise VdtParamError(
+			name_or_msg=f"Param 'default' not expected. {usage}",
+			value=specString
+		)
+
+	optionsEnumKey = "optionsEnum"
+	if optionsEnumKey not in kwargs:
+		raise VdtParamError(
+			name_or_msg=f"Param '{optionsEnumKey}' missing. {usage}",
+			value=specString
+		)
+	optionsEnumVal = kwargs[optionsEnumKey]
+	if not isinstance(optionsEnumVal, str):
+		raise VdtParamError(
+			name_or_msg=(
+				f"Param '{optionsEnumKey}' should have a string value"
+				f" but got {type(optionsEnumVal)}. {usage}"),
+			value=specString
+		)
+	availableEnums = dict(featureFlagEnums.getAvailableEnums())
+	if optionsEnumVal not in availableEnums:
+		raise VdtParamError(
+			name_or_msg=(
+				f"Param '{optionsEnumKey}' should be an enum defined in featureFlagEnums,"
+				f" but was {optionsEnumVal}. Currently available: {availableEnums.keys()} "
+			),
+			value=specString
+		)
+	OptionsEnumClass: enum.EnumMeta = availableEnums[optionsEnumVal]
+	behaviorOfDefaultKey = "behaviorOfDefault"
+	if behaviorOfDefaultKey not in kwargs:
+		raise VdtParamError(
+			name_or_msg=f"Param '{behaviorOfDefaultKey}' missing. {usage}",
+			value=specString
+		)
+	behaviorOfDefaultVal = kwargs[behaviorOfDefaultKey]
+	if not isinstance(behaviorOfDefaultVal, str):
+		raise VdtParamError(
+			name_or_msg=(
+				f"Param '{behaviorOfDefaultKey}' should have a string value"
+				f" but got {type(behaviorOfDefaultVal)}. {usage}"),
+			value=specString
+		)
+	behaviorOfDefaultVal = behaviorOfDefaultVal.upper()
+	try:
+		OptionsEnumClass[behaviorOfDefaultVal]
+	except KeyError:
+		raise VdtParamError(
+			name_or_msg=(
+				f"Param '{behaviorOfDefaultKey}' should be one of: {[o.name for o in OptionsEnumClass]}"
+				f" but was {behaviorOfDefaultVal}. {usage}"
+			),
+			value=specString
+		)
+	if len(kwargs) != 2:
+		raise VdtParamError(
+			name_or_msg=(
+				f"Unexpected number of params."

```

Comment:
No need for f-string here
```suggestion
				"Unexpected number of params."
```